frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-builder/__tests__/switch-to-form-validation-utils-data.ts

```diff
@@ -0,0 +1,46 @@
+import * as _ from 'lodash';
+import { apiVersionForModel } from '@console/internal/module/k8s';
+import { PipelineModel } from '../../../../models';
+import { PipelineKind } from '../../../../types/pipeline';
+import { pipelineBuilderYAMLSchema } from '../switch-to-form-validation-utils';
+import { PipelineBuilderFormValues } from '../types';
+
+export const yamlDataBasicPassState: PipelineKind = {
+  apiVersion: apiVersionForModel(PipelineModel),
+  kind: PipelineModel.kind,
+  metadata: {
+    name: 'initial-pipeline',
+    namespace: 'ns',
+  },
+  spec: {
+    params: [],
+    resources: [],
+    workspaces: [],
+    tasks: [],
+    finally: [],
+  },
+};
+
+export const updateYAML = (path, data) =>
+  path ? _.set(_.cloneDeep(yamlDataBasicPassState), path, data) : yamlDataBasicPassState;
+
+export const withFormData = (formData: PipelineBuilderFormValues, path, data) =>
+  pipelineBuilderYAMLSchema(formData).validate(updateYAML(path, data), {
+    abortEarly: false,
+    strict: true,
+  });
+
+export const hasError = (yupPath: string, errorMessage: string) => (error) => {
+  if (!error?.inner) {
+    // Not a yup validation object, do a bad comparison so the test echos it
+    expect(error).toBe('Not a Yup Error, see following error message for the actual error.');
+    return;
+  }
+
+  const errors: { path: string; message: string }[] = error.inner.map((err) => ({
+    path: err.path,
+    message: err.message,
+  }));
+  const expectedError = { path: yupPath, message: expect.stringContaining(errorMessage) };
```

Comment:
the error message also has details about the received invalid type, which isn't necessary to match.

Reply:
Because it's difficult to do? Tests shouldn't really be givin a lot of flexibility. I think the wise thing to do is to test the string exactly so we know we are getting the expected error.